pkg/config/remote/service/subscriber.go

```diff
@@ -46,57 +40,42 @@ func NewSubscriber(product pbgo.Product, refreshRate time.Duration, callback Sub
 	}
 }
 
-// NewGRPCSubscriber returns a new gRPC stream based subscriber.
-func NewGRPCSubscriber(product pbgo.Product, callback SubscriberCallback) (context.CancelFunc, error) {
-	creds := credentials.NewTLS(&tls.Config{
-		InsecureSkipVerify: true,
+// NewChanSubscriber returns a new subscriber that will put the ConfigResponse objects onto a provided channel.
+func NewChanSubscriber(product pbgo.Product, refreshRate time.Duration, channel chan *pbgo.ConfigResponse) *Subscriber {
+	return NewSubscriber(product, refreshRate, func(config *pbgo.ConfigResponse) error {
+		select {
+		case channel <- config:
+			return nil
+		default:
+			return errors.New("failed to put config onto channel")
+		}
 	})
+}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	conn, err := grpc.DialContext(
-		ctx,
-		fmt.Sprintf(":%v", config.Datadog.GetInt("cmd_port")),
-		grpc.WithTransportCredentials(creds),
-	)
-	if err != nil {
-		cancel()
-		return nil, err
-	}
-
-	agentClient := pbgo.NewAgentSecureClient(conn)
+// NewGRPCSubscriber returns a new gRPC stream based subscriber.
+func NewGRPCSubscriber(product pbgo.Product, callback SubscriberCallback) (context.CancelFunc, error) {
+	currentConfigSnapshotVersion := uint64(0)
+	client := tuf.NewDirectorPartialClient()
 
-	token, err := security.FetchAuthToken()
+	cas, err := newCoreAgentStream()
 	if err != nil {
-		cancel()
-		err = fmt.Errorf("unable to fetch authentication token: %w", err)
-		log.Infof("unable to establish stream, will possibly retry: %s", err)
-		return nil, err
+		cas.cancel()
+		return nil, fmt.Errorf("failed to create core agent stream: %w", err)
 	}
-
-	currentConfigSnapshotVersion := uint64(0)
-	client := tuf.NewDirectorPartialClient()
+	defer cas.streamCancel()
 
 	go func() {
 		log.Debug("Waiting for configuration from remote config management")
 
-		streamCtx, streamCancel := context.WithCancel(
-			metadata.NewOutgoingContext(ctx, metadata.MD{
-				"authorization": []string{fmt.Sprintf("Bearer %s", token)},
-			}),
-		)
-		defer streamCancel()
-
-		log.Debug("Processing response from remote config management")
-
 		for {
 			request := pbgo.SubscribeConfigRequest{
 				CurrentConfigSnapshotVersion: currentConfigSnapshotVersion,
 				Product:                      product,
 			}
-			stream, err := agentClient.GetConfigUpdates(streamCtx, &request)
+			stream := cas.getStream()
```

Comment:
Do we want to use `cas.stream` here? I'm confused about how `connect()` and `getStream()` relate to each other.